src/components/view/pages/main/sorting/sortingView.ts

```diff
@@ -0,0 +1,192 @@
+import './sortingView.scss';
+import {View} from "../../../view";
+import {Controller} from "../../../../controller/controller";
+import {Router} from "../../../../router/router";
+import {ProductViewDataType} from "../../../../types/sortDataType";
+import {PaginationPerPageView} from "../../../common/pagination/paginationPerPageView";
+
+export class SortingFiltersView extends View<ProductViewDataType> {
+    private readonly sortOptions: { [key: string]: string } = {
+        '': 'Recommended',
+        'name': 'Name',
+        'price-asc': 'Price ascending',
+        'price-desc': 'Price descending',
+        'stock-asc': 'Stock ascending',
+        'stock-desc': 'Stock descending',
+    }
+
+    protected views = {
+        paginationPerPage: new PaginationPerPageView(),
+    }
+
+    public render(data: ProductViewDataType): string {
+        let colorFilter = '';
+        if (data.filters.colors) {
+            // language=HTML
+            colorFilter = data.filters.colors.map((c) => {
+                return `
+                  <div class="selected-filters__item selected-item">
+                    <div class="selected-item__name">${c}</div>
+                    <div class="selected-item__remove-btn" data-params="colors" data-value="${c}"></div>
+                  </div>`;
+            }).join('')
+        }
+        let collectionFilter = '';
+        if (data.filters.collections) {
+            // language=HTML
+            collectionFilter = data.filters.collections.map((c) => {
+                return `
+                  <div class="selected-filters__item selected-item">
+                    <div class="selected-item__name">${c}</div>
+                    <div class="selected-item__remove-btn" data-params="collections" data-value="${c}"></div>
+                  </div>`;
+            }).join('')
+        }
```

Comment:
This is a good place to use `? :`
`const collectionFilter = collections.length ? collections.map(...) : ''`
Please use `el` or readable name in `arr.map()`